common/beatmap_utils.py

```diff
@@ -0,0 +1,472 @@
+from __future__ import annotations
```

Comment:
All of this is copy/pasta from these two score-service modules:

- https://github.com/osuAkatsuki/score-service/blob/6a8c37688a10e20dbdb5ab022cf7201edff6f3b2/app/usecases/beatmap.py#L175
- https://github.com/osuAkatsuki/score-service/blob/6a8c37688a10e20dbdb5ab022cf7201edff6f3b2/app/models/beatmap.py#L1